api/app/controllers/api/v1/offers_controller.rb

```diff
@@ -1,31 +1,396 @@
+        params[:offers].each do |id|
+          offer = Offer.find(id)
+          offer.increment!(:nag_count, 1)
+          if ENV['RAILS_ENV'] != 'test'
+            CpMailer.nag_email(offer.format, "#{ENV["domain"]}#{offer[:link]}").deliver_now!
+          end
+        end
+        render json: {message: "You've sent the nag emails."}
+      rescue Errno::ECONNREFUSED
+        render status: 404, json: {message: "Connection Refused."}
+      end
+    end
+
+    '''
+      Nag Mails (admin)
+    '''
+    def can_nag_instructor
+      invalid = []
+      params[:offers].each do |offer_id|
+        offer = Offer.find(offer_id)
+        instructors = offer.format[:instructors]
+        if !(["None", "Created"].include? offer[:ddah_status]) || instructors.length == 0
+          invalid.push(offer[:id])
+        end
+      end
+      if invalid.length > 0
+        render status: 404, json: {invalid_offers: invalid}
+      end
+    end
 
-        # POST /create_offer
-        def create
-            # TODO
-            render_error('')
+    def send_nag_instructor
+      begin
+        params[:offers].each do |id|
+          offer = Offer.find(id)
+          instructors = offer.format[:instructors]
+          offer.increment!(:ddah_nag_count, 1)
+          if ENV['RAILS_ENV'] != 'test'
+            instructors.each do |instructor|
+              CpMailer.instructor_nag_email(offer.format, instructor).deliver_now!
+            end
+          end
         end
+        render status: 200, json: {message: "You've successfully sent out all the nag emails."}
+      rescue Errno::ECONNREFUSED
+        render status: 404, json: {message: "Connection Refused."}
+      end
+    end
 
-        # GET /active_offer
-        def active_offer
-            # TODO
-            render_error('')
+
+    def combine_contracts_print
+      offers = []
+      params[:offers].each do |offer_id|
+        if params[:update]
+          update_print_status(offer_id)
         end
+        offer = Offer.find(offer_id)
+        offers.push(offer)
+      end
+      # create a pdf concatenation of all the offers using CombinePDF
+      out_pdf = CombinePDF.new
+      offers.each do |offer|
+        rendered = render_to_string pdf: "contract", inline: get_contract_html(offer, true), encoding: "UTF-8"
+        out_pdf << CombinePDF.parse(rendered)
+      end
+      send_data out_pdf.to_pdf, filename: "contracts.pdf", disposition: "inline"
+    end
+
+    def get_contract
+      rendered = get_contract_html(Offer.find(params[:offer_id]))
+      render pdf: "contract", inline: rendered
+    end
+
+    def get_contract_student
+      rendered = get_contract_html(Offer.find(params[:offer_id]))
+      render pdf: "contract", inline: rendered
+    end
+
+    def get_contract_student_html
+      rendered = get_contract_html(Offer.find(params[:offer_id]))
+      render inline: rendered
+    end
+
+    def set_status_student
+      if params[:status] == "accept" || params[:status]== "reject"
```

Comment:
A signature is also required if the student accepts, I think its better to split this into two actions.